### PR TITLE
Feat/gauge veboost + test

### PR DESCRIPTION
**Pull request by skellet0r**
Test the veboost is called correctly and update the gauge

### CODE REVIEW

File: contracts/LiquidityGauge.vy [hunk 1]
```diff
@@ -90,6 +93,7 @@ MINTER: constant(address) = 0xd061D61a4d941c39E5453435B6345Dc261C2fcE0
 CRV: constant(address) = 0xD533a949740bb3306d119CC777fa900bA034cd52
 VOTING_ESCROW: constant(address) = 0x5f3b5DfEb7B28CDbD7FAba78963EE202a494e2A2
 GAUGE_CONTROLLER: constant(address) = 0x2F50D538606Fa9EDD2B11E2446BEb18C9D5846bB
+VEBOOST_PROXY: constant(address) = 0x0000000000000000000000000000000000000000
```
Comment (skellet0r) [inline contracts/LiquidityGauge.vy:96]: This needs to be updated to the correct veboost proxy address

File: tests/fixtures/deployments.py [hunk 2]
```diff
@@ -232,15 +237,20 @@ def minter(alice, crv, pm, gauge_controller):
...
+def gauge_implementation(
+    alice, LiquidityGauge, minter, crv, voting_escrow, gauge_controller, mock_veboost_proxy
+):
     source = LiquidityGauge._build["source"]
     old_addrs = [
         "0xd061D61a4d941c39E5453435B6345Dc261C2fcE0",  # minter
         "0xD533a949740bb3306d119CC777fa900bA034cd52",  # crv
         "0x5f3b5DfEb7B28CDbD7FAba78963EE202a494e2A2",  # voting escrow
         "0x2F50D538606Fa9EDD2B11E2446BEb18C9D5846bB",  # gauge controller
+        "0x0000000000000000000000000000000000000000",  # veboost proxy
```
Comment (skellet0r) [inline tests/fixtures/deployments.py:249]: This should be updated as well so tests don't break

File: contracts/LiquidityGauge.vy (hunk 1 above)
Comment (iamdefinitelyahuman) [reply contracts/LiquidityGauge.vy:96]: [`0x8E0c00ed546602fD9927DF742bbAbF726D5B0d16`](https://etherscan.io/address/0x8e0c00ed546602fd9927df742bbabf726d5b0d16#code)

File: tests/gauge/test_boost_delegation.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+def test_boost_delegation_set_in_addr_provider(alice, gauge, mock_veboost_proxy):
+    mock_veboost_proxy.set_delegation(ETH_ADDRESS, {"from": alice})
+
+    tx = gauge.user_checkpoint(alice, {"from": alice})
+    subcall = next(
+        (tx for tx in tx.subcalls if tx["function"] == "adjusted_balance_of(address)"), {}
+    )
+
+    assert subcall["return_value"][0] == 69
```
Comment (skellet0r) [inline tests/gauge/test_boost_delegation.py:24]: here we aren't testing anything other than the fact that the proxy is called. And for now we know that two behaviors exist, the first is if the proxy points to the veboost contract we will get the adjusted veboost, the second being if the delegation is not set, we should get the plain voting escrow balance of a user.

Those however are implementation details of the voting escrow boost contract itself.

**Review by iamdefinitelyahuman — APPROVED** [review]

**Pull request merged**